src/Subscriptions/StorageManager.php

```diff
@@ -110,8 +111,42 @@ public function storeSubscriber(Subscriber $subscriber, string $topic): void
         }
     }
 
+    /**
+     * Delete subscriber.
+     *
+     * @param  string  $channel
+     * @return \Nuwave\Lighthouse\Subscriptions\Subscriber|null
+     */
     public function deleteSubscriber(string $channel): ?Subscriber
     {
-        return $this->cache->pull(self::SUBSCRIBER_KEY.".{$channel}");
+        $key = self::SUBSCRIBER_KEY.".{$channel}";
+        $this->removeSubscriberFromTopic($key);
+
+        return $this->cache->pull($key);
+    }
+
+    /**
+     * Remove the subscriber from the topic they are subscribed to.
+     *
+     * @param  string $subscriberKey
+     */
+    protected function removeSubscriberFromTopic(string $subscriberKey)
+    {
+        $subscriber = $this->cache->get($subscriberKey);
+
+        if (!$subscriber || !$subscriber->topic) {
+            return;
+        }
```

Comment:
```suggestion
    protected function removeSubscriberFromTopic(Subscriber $subscriber)
    {
        if (empty($subscriber->topic)) {
            return;
        }
```

Considering my changes in the `deleteSubscriber` method.